driver.go

```diff
@@ -325,23 +386,35 @@ func (d *Driver) RecoverTask(handle *drivers.TaskHandle) error {
 	}
 	d.logger.Debug("Checking for recoverable task", "task", handle.Config.Name, "taskid", handle.Config.ID, "container", taskState.ContainerID)
 
-	inspectData, err := d.podman.ContainerInspect(d.ctx, taskState.ContainerID)
-	if errors.Is(err, api.ContainerNotFound) {
-		d.logger.Debug("Recovery lookup found no container", "task", handle.Config.ID, "container", taskState.ContainerID, "error", err)
-		return nil
-	} else if err != nil {
-		d.logger.Warn("Recovery lookup failed", "task", handle.Config.ID, "container", taskState.ContainerID, "error", err)
-		return nil
+	// TODO: It can be created by any podman client, so we need to iterate over all podman clients and find that container back...
+	var podmanClientName string
+	var taskPodmanClient *api.API
+	var inspectData api.InspectContainerData
+	var err error
+	for name, podmanClient := range d.podmanClients {
+		inspectData, err = podmanClient.ContainerInspect(d.ctx, taskState.ContainerID)
+		if err == nil {
+			podmanClientName = name
+			taskPodmanClient = podmanClient
+			break
+		} else if errors.Is(err, api.ContainerNotFound) {
+			d.logger.Debug("Recovery lookup found no container", "task", handle.Config.ID, "container", taskState.ContainerID, "with podman client", name, "error", err)
+		} else if err != nil {
+			d.logger.Warn("Recovery lookup failed", "task", handle.Config.ID, "container", taskState.ContainerID, "with podman client", name, "error", err)
+			// TODO: Maybe not return here?
+			return nil
```

Comment:
If you're getting an error other than `ContainerNotFound` there's a communications issue. We could return the error here or clear `err` if the container is found later.